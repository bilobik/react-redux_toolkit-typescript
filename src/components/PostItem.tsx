import React, { FC } from "react";
import { IPost } from "../models/IPost";

interface PostItemProps {
  post: IPost;
  remove: (post: IPost) => void;
  update: (post: IPost) => void;
}
const PostItem: FC<PostItemProps> = ({ post, remove, update }) => {
  const handleRemove = (event: React.MouseEvent) => {
    event.stopPropagation();
    remove(post);
  };

  const handleUpdate = (event: React.MouseEvent) => {
    event.stopPropagation();
    const postTitle = prompt();
    const o = { ...post, title: postTitle, body: postTitle } as IPost;
    update(o);
  };

  return (
    <div className="post" key={post.id} onClick={handleUpdate}>
      <h3>{post.title}</h3>
      <p>{post.body}</p>
      <button onClick={handleRemove}>Delete</button>
    </div>
  );
};

export default PostItem;
