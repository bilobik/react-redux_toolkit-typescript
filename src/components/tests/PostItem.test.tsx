import React from 'react';
import { render, screen } from '@testing-library/react';
import PostItem from '../PostItem';
import { IPost } from '../../models/IPost';

test('renders post', () => {
    const post: IPost = {
        id: 1,
        title: 'Foo',
        body: 'Bar'
    }
  render(<PostItem post={post} remove={()=>{}} update={() => {}} />);
  const titleElement = screen.getByText(/Foo/i);
  const postBodyElement = screen.getByText(/Bar/i);
  expect(titleElement).toBeInTheDocument();
  expect(postBodyElement).toBeInTheDocument();
  //screen.debug();
  expect(titleElement).toMatchSnapshot();
});
