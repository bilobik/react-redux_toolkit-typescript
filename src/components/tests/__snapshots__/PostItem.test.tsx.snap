// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders post 1`] = `
<h3>
  Foo
</h3>
`;
